corehq/apps/reports/standard/deployments.py

```diff
@@ -318,22 +325,32 @@ def process_rows(self, users, fmt_for_export=False):
                 last_builds = reporting_metadata.get('last_builds')
                 if last_builds:
                     last_build = self.get_data_for_app(last_builds, self.selected_app_id)
+                if device and is_commcare_user:
+                    device_app_meta = self.get_data_for_app(device.get('app_meta'), self.selected_app_id)
             else:
                 last_sub = reporting_metadata.get('last_submission_for_user', {})
                 last_sync = reporting_metadata.get('last_sync_for_user', {})
                 last_build = reporting_metadata.get('last_build_for_user', {})
+                if is_commcare_user:
+                    app_id = None
+                    if last_build:
+                        app_id = last_build.get('app_id')
+                    if device and app_id:
+                        device_app_meta = device.get('app_meta')
+                        if device_app_meta:
+                            device_app_meta = self.get_data_for_app(device_app_meta, app_id)
```

Comment:
is this equivalent to
```
if is_commcare_user and last_build.get('app_id') and device and device.get('app_meta'):
    device_app_meta = self.get_data_for_app(device_app_meta, last_build.get('app_id'))
```
if that looks like a long chain can get `device_app_meta = device.get('app_meta')` and use that instead.
`last_build` is a dict by default, so `if last_build` is redundant.

Reply:
updated